### PR TITLE
Fix - Can't subsequently run actions for different objects from dashboards

### LINKED ISSUE #33084: Can't subsequently run actions for different objects from dashboards
Epic: #32320

### Description

Broken by refactoring in #32671.

### Steps to reproduce 

1. Create a model
2. Enable basic actions in it
3. Create a dashboard
4. Add an ID filter to the dashboard
5. Add an action button to the dashboard, use basic update action of created model
6. Connect the ID filter as a parameter to the action button
7. Set ID filter to 1
8. Click the action button to run it
9. Notice the update modal contains data for object with id = 1
10. Close the modal
11. Set ID filter to 2
12. Click the action button to run it

Expected result: the update modal contains data for object with id = 2
Actual result: the update modal contains data for object with id = 1

### Demo

https://github.com/metabase/metabase/assets/6830683/7c26273e-4176-430d-a846-3313bce18557

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #33084

### Description

State hook does not get remounted after [refactoring](https://github.com/metabase/metabase/pull/3267), so we need to make sure it properly refreshes the state.

### How to verify

Follow reproduction steps in original report.

#### Before

https://github.com/metabase/metabase/assets/6830683/e8fbc6cd-fb14-4248-a2c8-e9414225827b

#### After

https://github.com/metabase/metabase/assets/6830683/ada5b76d-9b0e-4062-9b0b-f5b38110a97b

### CODE REVIEW

File: e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js [hunk 1]
```diff
@@ -987,6 +992,55 @@ describe(
...
+
+        createDashboardWithActionButton({
+          actionName,
+          idFilter: true,
+        });
+
+        filterWidget().click();
+        addWidgetStringFilter("5");
+
+        cy.findByRole("button", { name: actionName }).click();
```
Comment (WiNloSt) [inline e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js:1013]: This is the same as
```suggestion
        cy.button(actionName).click();
```

while `cy.button` is far more established. 500+ results vs 66 results of `findByRole("button"...` inside `e2e` folder.

File: e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js [hunk 2]
```diff
@@ -987,6 +992,55 @@ describe(
...
+          cy.findByPlaceholderText("Score").should("have.value", "30");
+
+          cy.icon("close").click();
+        });
+
+        filterWidget().click();
+        popover().find("input").first().type("{backspace}10");
+        cy.button("Update filter").click();
+
+        cy.findByRole("button", { name: actionName }).click();
```
Comment (WiNloSt) [inline e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js:1031]: This is the same as
```suggestion
        cy.button(actionName).click();
```

**Review by WiNloSt — APPROVED** [review]
Looks correct to me. `prefetchValues` will be called again once we pass a new `prefetchValues` for a different entity.

File: e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js (hunk 1 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js]: 35990d6777a6694eedb473abc58847de3249d2e7

File: e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js (hunk 2 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js]: 35990d6777a6694eedb473abc58847de3249d2e7

**Pull request merged**